src/scheduler/analysis_status.py

```diff
@@ -52,13 +54,23 @@ def _update_current_analysis(self, fw_object: FileObject):
                 set(updated_dict['files_to_analyze'])
             )
             updated_dict['total_files_count'] += len(new_files)
+            updated_dict['total_files_with_duplicates'] += 1
             updated_dict['files_to_unpack'] = list(set(updated_dict['files_to_unpack']).union(new_files))
             if fw_object.uid in updated_dict['files_to_unpack']:
                 updated_dict['files_to_unpack'].remove(fw_object.uid)
                 updated_dict['files_to_analyze'].append(fw_object.uid)
                 updated_dict['unpacked_files_count'] += 1
             self.currently_running[parent] = updated_dict
 
+    def update_post_analysis(self, fw_object: FileObject, plugin: str):
+        for parent in self._find_currently_analyzed_parents(fw_object):
+            self.currently_running_lock.acquire()
```

Comment:
`with self._get_lock():`

Reply:
This should then be acquired outside the loop. I don't see why it should be release in the interim.

Reply:
changed